### ISSUE TITLE
Support differentiating between static and dynamic resource groups

### ISSUE THREAD

**Issue by ddanielr** [apache/accumulo#5162]
In the past accumulo has tracked dead tservers. 

For resource groups that have dynamic scaling workloads, there should be an option to disable this tracking of "dead" servers. 

We probably don't want to remove this functionality entirely because there might be a static group of tservers which could have issues and failures for those machines should be tracked.

**Comment by dlmarion** [comment]
Another way to approach this would be to remove the tablet servers ZooKeeper entry on a normal shutdown. I believe that the tserver did not do this in earlier (< 4.0) versions because the configuration when using cluster.yaml because the deployment was meant to be static.

In fact, this is a larger issue when a users uses something like Kubernetes. It's possible that old server paths might linger for all the server processes. We may want to consider removing the server paths from ZooKeeper for all server types when performing a normal shutdown.

**Comment by dlmarion** [comment]
> Another way to approach this would be to remove the tablet servers ZooKeeper entry on a normal shutdown. 

In the ServiceLock.unlock code (below) in the case of Compactors, Scan Servers, and TabletServers `path` is `/accumulo/<instanceId>/(compactors|sservers|tservers)/<resourceGroup>/<address>` and `localLock` is the ephemeral node that is created. The code is currently only deleting the ephemeral node.

We could modify `ServiceLock.unlock` to accept a boolean, which when `true` could perform a recursive delete on `path`. We would only set it to `true` in the case of Compactors, ScanServers, and TabletServers when ServiceLock.unlock is called. This would mean that the LiveTServerSet running in the Manager would not find any TabletServers without locks when they are shut down gracefully. Graceful shutdown is being addressed in #5193.

https://github.com/apache/accumulo/blob/e5370d1d5cd436d86170d16b0e8c0986d3275e15/core/src/main/java/org/apache/accumulo/core/lock/ServiceLock.java#L531-L558

**Comment by dlmarion** [comment]
I created #5226 to suppress the display of dead servers for configured resource groups in the Monitor. I think this is the simplest approach at this point. My earlier comments about removing the lock in ZooKeeper may not fully resolve the issue you raise.

**Comment by ddanielr** [comment]
> I created #5226 to suppress the display of dead servers for configured resource groups in the Monitor. I think this is the simplest approach at this point. My earlier comments about removing the lock in ZooKeeper may not fully resolve the issue you raise.

Agreed, I think that #5226 is a good solution for this issue as it's not tied to the cluster.yaml definition and also covers the situation where the user might want to shut down tservers faster than the "graceful shutdown" process allows and would end up generating dead tserver locks.
